setup.py

```diff
@@ -1,76 +0,0 @@
-import os
```

Comment:
I am afraid we cannot delete the setup.py, due to readthedocs (which, for the time being, stays as our documentation generation). 

We can still significantly reduce it:
https://packaging.python.org/en/latest/guides/modernize-setup-py-project/#how-to-handle-additional-build-time-dependencies

Reply:
I don't follow you. You linked the buid system. It's not related to sphinx.

I tried to run the [doc build command](https://www.sphinx-doc.org/en/master/usage/quickstart.html#running-the-build) ``sphinx-build -M html sourcedir outputdir`` and it properly generated the doc

However, I realized that I needed to add ``sphinx>=5.0.0`` to the optional dependencies, which should have been included in setup.py.

Reply:
The problem is in [readthedocs](https://about.readthedocs.com/) website, which we are using to host PyonFX's [doc](https://pyonfx.rtfd.io/). The website fetches the updated repo through a webhook, and builds the doc on its own (that's why we don't have generated doc here on github).

But I remember that when trying to build, it couldn't fetch the os-specific libraries (but I'm not sure if it will be doing the same thing now, so I would like to give it a try with your PR and see what errors we get).

Reply:
It should be fine since you specified ``dev`` https://github.com/CoffeeStraw/PyonFX/blob/101797c6015469ea19b9ea08367369d14394c86d/readthedocs.yml#L12

Reply:
Let's give it a try :)